### ISSUE TITLE
uwsgi does not find pcre

### ISSUE THREAD

**Issue by bbarker** [NixOS/nixpkgs#32825]
## Issue description

When running uwsgi on a project, I see the following, with the relevant line being `!!! no internal routing support, rebuild with pcre support !!!`:                                                                                                                                                                                                                 
```
[nix-shell:~/workspace/arxiv-zero]$  uwsgi --http-socket :8000 -w wsgi -t 3000 --processes 8 --threads 1 -M --async 100 --ugreen --manage-script-name                                                                                                                                                                                                                        
*** Starting uWSGI 2.0.15 (64bit) on [Mon Dec 18 21:33:36 2017] ***                                                                                                                                                                                                                                                                                                          
compiled with version: 6.4.0 on 04 December 2017 13:31:03                                                                                                                                                                                                                                                                                                                    
os: Linux-4.10.0-38-generic #42~16.04.1-Ubuntu SMP Tue Oct 10 16:32:20 UTC 2017                                                                                                                                                                                                                                                                                              
nodename: beb82dell0                                                                                                                                                                                                                                                                                                                                                         
machine: x86_64                                                                                                                                                                                                                                                                                                                                                              
clock source: unix                                                                                                                                                                                                                                                                                                                                                           
detected number of CPU cores: 8                                                                                                                                                                                                                                                                                                                                              
current working directory: /home/brandon/workspace/arxiv-zero                                                                                                                                                                                                                                                                                                                
detected binary path: /home/brandon/workspace/arxiv-zero/venv/bin/uwsgi                                                                                                                                                                                                                                                                                                      
!!! no internal routing support, rebuild with pcre support !!!                                                                                                                                                                                                                                                                                                               
your memory page size is 4096 bytes                                                                                                                                                                                                                                                                                                                                          
 *** WARNING: you have enabled harakiri without post buffering. Slow upload could be rejected on post-unbuffered webservers ***                                                                                                                                                                                                                                              
detected max file descriptor number: 1048576                                                                                                                                                                                                                                                                                                                                 
- async cores set to 100 - fd table size: 1048576                                                                                                                                                                                                                                                                                                                            
lock engine: pthread robust mutexes                                                                                                                                                                                                                                                                                                                                          
thunder lock: disabled (you can enable it with --thunder-lock)                                                                                                                                                                                                                                                                                                               
initializing 100 uGreen threads with stack size of 262144 (256 KB)                                                                                                                                                                                                                                                                                                           
uwsgi socket 0 bound to TCP address :8000 fd 3                                                                                                                                                                                                                                                                                                                               
Python version: 3.6.3 (default, Oct  3 2017, 07:47:49)  [GCC 6.4.0]                                                                                                                                                                                                                                                                                                          
Python main interpreter initialized at 0x218aeb0                                                                                                                                                                                                                                                                                                                             
python threads support enabled                                                                                                                                                                                                                                                                                                                                               
your server socket listen backlog is limited to 100 connections                                                                                                                                                                                                                                                                                                              
your mercy for graceful operations on workers is 60 seconds                                                                                                                                                                                                                                                                                                                  
mapped 9828576 bytes (9598 KB) for 800 cores                                                                                                                                                                                                                                                                                                                                 
*** Operational MODE: preforking+async ***                                                                                                                                                                                                                                                                                                                                   
WSGI app 0 (mountpoint='') ready in 1 seconds on interpreter 0x218aeb0 pid: 6897 (default app)                                                                                                                                                                                                                                                                               
*** uWSGI is running in multiple interpreter mode ***                                                                                                                                                                                                                                                                                                                        
spawned uWSGI master process (pid: 6897)                                                                                                                                                                                                                                                                                                                                     
spawned uWSGI worker 1 (pid: 6900, cores: 100)                                                                                                                                                                                                                                                                                                                               
spawned uWSGI worker 2 (pid: 6901, cores: 100)                                                                                                                                                                                                                                                                                                                               
spawned uWSGI worker 3 (pid: 6902, cores: 100)                                                                                                                                                                                                                                                                                                                               
spawned uWSGI worker 4 (pid: 6903, cores: 100)                                                                                                                                                                                                                                                                                                                               
spawned uWSGI worker 5 (pid: 6904, cores: 100)                                                                                                                                                                                                                                                                                                                               
spawned uWSGI worker 6 (pid: 6905, cores: 100)                                                                                                                                                                                                                                                                                                                               
spawned uWSGI worker 7 (pid: 6906, cores: 100)                                                                                                                                                                                                                                                                                                                               
spawned uWSGI worker 8 (pid: 6907, cores: 100)        
```

### Steps to reproduce

In a nix-shell expression, put `pcre` a a buildInput, and install `uwsgi` either as a buildInput (not expected to work) or via pip after the nix-shell has launched (I expected this to work). [Run uwsgi on a simple application](http://uwsgi-docs.readthedocs.io/en/latest/WSGIquickstart.html#the-first-wsgi-application) and observe the output as above. 

## Technical details

```
$ nix-shell -p nix-info --run "nix-info -m"                                                                                                                                                                                                                                                                                                                                  
warning: Nix search path entry ‘/nix/var/nix/profiles/per-user//nixenv/brandon/channels/’ does not exist, ignoring                                                                                                                                                                                                                                                           
warning: Nix search path entry ‘/nix/var/nix/profiles/per-user//nixenv/brandon/channels/’ does not exist, ignoring                                                                                                                                                                                                                                                           
 - system: `"x86_64-linux"`                                                                                                                                                                                                                                                                                                                                                  
 - host os: `Linux 4.10.0-38-generic, Ubuntu, 16.04.3 LTS (Xenial Xerus)`                                                                                                                                                                                                                                                                                                    
 - multi-user?: `no`                                                                                                                                                                                                                                                                                                                                                         
 - sandbox: `no`                                                                                                                                                                                                                                                                                                                                                             
 - version: `nix-env (Nix) 1.11.15`                                                                                                                                                                                                                                                                                                                                          
 - channels(brandon): `"nixos-18.03pre121706.b9347167822, nixpkgs-18.03pre121653.a88146d308f"`                                                                                                                                                                                                                                                                               
warning: Nix search path entry ‘/nix/var/nix/profiles/per-user//nixenv/brandon/channels/’ does not exist, ignoring                                                                                                                                                                                                                                                           
 - nixpkgs: `/home/brandon/.nix-defexpr/channels/nixpkgs`                                                                                                                                                                                                                                                                                                                    
                                                                                                                                                                                                                                                                                                                                                                             
```

**Comment by bbarker** [comment]
Here's my example shell file, which can probably be trimmed a bit:

```
with import <nixpkgs> { config = {
  packageOverrides = pkgs: {
    uwsgi = pkgs.uwsgi.override { xlibs = null; }; # how to make this enable pcre?
  };
};};
with pkgs.python36Packages;
stdenv.mkDerivation {
  name = "impurePythonEnv";
  buildInputs = [
    # these packages are required for virtualenv and pip to work:
    #
    python36Full
    python36Packages.virtualenv
    python36Packages.pip
    # the following packages are related to the dependencies of your python
    # project.
    # In this particular example the python modules listed in the
    # requirements.tx require the following packages to be installed locally
    # in order to compile any binary extensions they may require.
    #
    gcc6
    mysql57
    ncurses # needed by uWSGI
    openssl
    pcre
    zlib
  ];
  src = null;
  shellHook = ''
    # set SOURCE_DATE_EPOCH so that we can use python wheels
    SOURCE_DATE_EPOCH=$(date +%s)
    virtualenv venv
    export PATH=$PWD/venv/bin:$PATH
    export PYTHONPATH=$PWD
    pip install -r requirements.txt
    export LD_LIBRARY_PATH=${mysql57}/lib:${gcc6.cc.lib}/lib:$LD_LIBRARY_PATH
    export FLASK_APP=app.py
    export FLASK_DEBUG=1
    source private_vars.sh
  '';
}

#
# Now you can run the following command to start the server:
#
# uwsgi --http-socket :8000 -w wsgi -t 3000 --processes 8 --threads 1 -M --async 100 --ugreen --manage-script-name
```

**Comment by orivej** [comment]
Could you try installing `(uwsgi.override { plugins = [ "python3" ]; })`?

**Comment by azazel75** [comment]
will this be merged into 18.09?
